sources/metricdog/src/main.rs

```diff
@@ -0,0 +1,121 @@
+#![deny(rust_2018_idioms, unused_imports)]
+
+/*!
+# Introduction
+
+Metricdog sends anonymous information about the health of a Bottlerocket host.
+It does so by sending key-value pairs as query params in an HTTP GET request.
+
+Metricdog also has the ability to check that a list of critical services is running.
+It does so using `systemctl` and reports services that are not healthy.
+
+# What it Sends
+
+### The standard set of metrics:
+
+* `sender`: the application sending the report.
+* `event`: the event that invoked the report.
+* `version`: the Bottlerocket version.
+* `variant`: the Bottlerocket variant.
+* `arch`: the machine architecture, e.g.'x86_64' or 'arm'.
+* `aws_region`: the AWS region the machine is running in.
+* `seed`: the seed value used to roll-out updates.
+* `version_lock`: the optional setting that controls Bottlerocket update selection.
+* `ignore_waves`: an update setting that allows hosts to update before their seed is reached.
+
+### Additionally, when `metricdog` sends a 'health ping', it adds:
+
+* `is_healthy`: true or false based on whether critical services are running.
+* `failed_services`: a list of critical services that have failed, if any.
+
+# Configuration
+
+Configuration is read from a TOML file, which is generated from Bottlerocket settings:
+
+```toml
+# the url to which metricdog will send metrics information
+metrics_url = "https://example.com/metrics"
+# whether or not metricdog will send metrics. opt-out by setting this to false
+send_metrics = true
+# a list of systemd service names that will be checked
+service_checks = ["apiserver", "containerd", "kubelet"]
+# the AWS region
+aws_region = "us-west-2"
+# the update wave seed
+seed = 1234
+# what version bottlerocket should stay on
+version_lock = "latest"
+# whether bottlerocket should ignore update roll-out timing
+ignore_waves = false
+```
+*/
+
+#![deny(rust_2018_idioms)]
+
+mod args;
+mod config;
+mod error;
+mod metricdog;
+#[cfg(test)]
+mod metricdog_test;
+#[cfg(test)]
+mod main_test;
+mod service_check;
+
+use crate::args::{Arguments, Command};
+use crate::config::Config;
+use crate::error::Result;
+use crate::metricdog::Metricdog;
+use crate::service_check::{ServiceCheck, SystemdCheck};
+use bottlerocket_release::BottlerocketRelease;
+use env_logger::Builder as EnvLogger;
+use log::error;
+use snafu::ResultExt;
+use std::process;
+use structopt::StructOpt;
+
+fn main() -> ! {
+    let args = Arguments::from_args();
+    EnvLogger::new()
```

Comment:
Is there a reason to veer away from simplelog like we use in all our other crates?  We're talking about changing our logger, but that's easier if we're consistent.

Reply:
At first I copied code from updog which panicked when systemd execute the program. Looking deeper now it appears maybe I could have instantiated a SimpleLogger(?) instead of TermLogger, but didn't figure that out at the time. Seems like simplelog has caused issues for us elsewhere as well.

`env_logger` is more popular, is highly user-configurable (at runtime), seems to have fewer foot-guns and seems like the closest thing to a de-facto standard (though there are way, way too many logger crates).

What I like about env_logger is that, as the user at runtime you can control the logging verbosity of even dependent crates. We should probably choose a logger and standardize on it, but I wouldn't choose simplelog for the standard. I'd be happy to replace simplelog with env_logger throughout the codebase (in a separate PR) if we want to get that done.

Reply:
env_logger does have benefits, but I don't think it can be set up with split stdout/stderr like we need.  simplelog makes that easy.  That actually raises a point I missed here; metricdog service files should have `StandardError=journal+console` so we see errors on console, otherwise it'd be harder to debug.

Reply:
Ugh, sadly `env_logger` won't work: https://docs.rs/env_logger/0.7.1/env_logger/struct.Builder.html#method.target

Reply:
Switched to simplelog in [c966f19](https://github.com/bottlerocket-os/bottlerocket/compare/b8b5c3ad92298db2c46262eb2b6c5b6ea3f60187..c966f19f71c044ce4a20c0a363245035f37b5737)